samples/azure-quantum/grover-sudoku/Grovers-sudoku-quantinuum.ipynb

```diff
@@ -0,0 +1,1425 @@
+        "        jobName=self.job_name(),\n",
+        "        nVertices=nVertices,\n",
+        "        bitsPerColor=self.bit_length(),\n",
+        "        emptySquareEdges1=empty_square_edges_1,\n",
+        "        emptySquareEdges2=empty_square_edges_2,\n",
+        "        startingNumberConstraints1=starting_number_constraints_1,\n",
+        "        startingNumberConstraints2=starting_number_constraints_2\n",
+        "    )\n",
+        "    print(job_msg(job))\n",
+        "    return job\n",
+        "\n",
+        "\n",
+        "Sudoku.job_name = job_name\n",
+        "Sudoku.submit_job_to_azure = submit_job_to_azure"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "Let's now run our code on Quantinuum targets. To check that it is compatible with Quantinuum machines, let us first run it on the API validator."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "qsharp.azure.target('quantinuum.hqs-lt-s1-apival')\n",
+        "\n",
+        "api_validation_jobs = list()\n",
+        "for puzzle in h11_puzzles:  # We'll be using the puzzles we found viable earlier!\n",
+        "    sudoku = Sudoku(puzzle)\n",
+        "    if not sudoku.count_empty_squares() >= 4:\n",
+        "        continue\n",
+        "    print(sudoku)\n",
+        "    job = sudoku.submit_job_to_azure()\n",
+        "    api_validation_jobs.append(job)\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "all_completed = True\n",
+        "for job in api_validation_jobs:\n",
+        "    (completed, _) = get_job(job)\n",
+        "    if not completed:\n",
+        "        all_completed = False\n",
+        "if all_completed:\n",
+        "    # Previous call would throw exception if invalid\n",
+        "    print(\"API validation succeeded!\")\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "After seeing that everything works smoothly, we will run on the H1-1 emulator.\n",
+        " \n",
+        "The H1-1 emulator can accurately model the noise of the H1-1 quantum machine, allowing us get good results while staying within our free credits. \n",
+        "To save on credits we only run the interesting examples of 5 unknowns. \n",
+        "One of the examples has a search space size of 1, essentially making it state preparation and measurement.\n",
+        " The other however, has a search space size of 4, causing us to access the Oracle.\n",
+        "\n",
+        "**Please note that this sample makes use of paid services on Azure Quantum. \n",
+        "The cost of running this sample with the provided parameters on Quantinuum in a free trial subscription is approximately 360EHQC (equivalent to 1125$). \n",
```

Comment:
We should double-check that this estimate is accurate and is phrased correctly; please double-check wording with PMs. Thanks!